File: 2024/Python/puzzle-input/puzzle-input.py
#!/usr/bin/env python3

import requests
from dotenv import load_dotenv
import os

load_dotenv()

cookie = os.getenv('PUZZLE_INPUT_COOKIE')

def get_puzzle_input(day):
    headers = {'Cookie': f'session={os.getenv("TOKEN")};'}
    response = requests.get(f'https://adventofcode.com/2024/day/{day}/input', headers=headers)
    if response.status_code == 200:
        return response.text
    else:
        response.raise_for_status()

def main():
    print(get_puzzle_input(1))

if __name__ == "__main__":
    main()

File: 2024/dotNet/AoC2024/PuzzleInput/PuzzleInput.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;

namespace PuzzleInput
{
    public class PuzzleInput
    {
        public PuzzleInput() { }
        public string GetPuzzleInput(int day, string token)
        {

            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri("https://adventofcode.com/2024/day/" + day + "/input");
                client.DefaultRequestHeaders.Add("Cookie", "session=" + token + ";");
                HttpResponseMessage response = client.GetAsync("").Result;
                response.EnsureSuccessStatusCode();
                string responseBody = response.Content.ReadAsStringAsync().Result;
                return responseBody;

            }
        }
    }

}
